packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts

```diff
@@ -76,28 +77,36 @@ export const onEvent = async (event: AWSCDKAsyncCustomResource.OnEventRequest):
       // determine if table needs replacement
       if (isKeySchemaModified(describeTableResult.Table.KeySchema!, tableDef.keySchema)) {
         console.log('Update requires replacement');
-
-        console.log('Deleting the old table');
-        await ddbClient.deleteTable({ TableName: event.PhysicalResourceId });
-        await retry(
-          async () => await doesTableExist(event.PhysicalResourceId!),
-          (res) => res === false,
-        );
-        console.log(`Table '${event.PhysicalResourceId}' does not exist. Deletion is finished.`);
-
-        const createTableInput = toCreateTableInput(tableDef);
-        const response = await createNewTable(createTableInput);
-        result = {
-          PhysicalResourceId: response.tableName,
-          Data: {
-            TableArn: response.tableArn,
-            TableStreamArn: response.streamArn,
-            TableName: response.tableName,
-            IsTableReplaced: true, // This value will be consumed by isComplete handler
-          },
-        };
-        log('Returning result', result);
-        return result;
+        if (tableDef.allowDestructiveGraphqlSchemaUpdates) {
+          if (describeTableResult.Table.DeletionProtectionEnabled === true) {
+            throw new Error('Table cannot be replaced when the deletion protection is enabled.');
+          }
+          console.log('Deleting the old table');
+          await ddbClient.deleteTable({ TableName: event.PhysicalResourceId });
+          await retry(
+            async () => await doesTableExist(event.PhysicalResourceId!),
+            (res) => res === false,
+          );
+          console.log(`Table '${event.PhysicalResourceId}' does not exist. Deletion is finished.`);
+
+          const createTableInput = toCreateTableInput(tableDef);
+          const response = await createNewTable(createTableInput);
+          result = {
+            PhysicalResourceId: response.tableName,
+            Data: {
+              TableArn: response.tableArn,
+              TableStreamArn: response.streamArn,
+              TableName: response.tableName,
+              IsTableReplaced: true, // This value will be consumed by isComplete handler
+            },
+          };
+          log('Returning result', result);
+          return result;
+        } else {
+          throw new Error(
```

Comment:
Do we have a test that verifies destructive operations throw an error if the flag isn't specified? If not, let's add one before we release, since that would be a catastrophic regression.

Reply:
Disregard, saw: https://github.com/aws-amplify/amplify-category-api/pull/2039/files#diff-0a311dff043ffa5e6e67f34000f2e8fee12b539efb486f9461c285b1ffe561ceR49